scipy/meson.build

```diff
@@ -590,6 +591,7 @@ subdir('sparse')
 subdir('stats')
 subdir('fft')
 subdir('io')
+subdir('optimize')
```

Comment:
Looking at build profiling for this change, looks good:

![image](https://github.com/user-attachments/assets/1a469701-77ab-424d-beac-e5b6351f1519)

HiGHS did get heavier I think, but that is what it is.